### ISSUE TITLE
RStudio build opens too many files at the same time

### ISSUE THREAD

**Issue by yurivict** [rstudio/rstudio#4543]
The FreeBSD package builder (poudriere) sets a limit on the number of open files, and this limit (```1000```) gets consistently exceeded. Currently, it has to be adjusted specifically for RStudio to ```4096``` in order to succeed.

Something is wrong with the build so that it opens too many files. Please fix this!

I am a maintainer of the FreeBSD port.

**Comment by ronblum** [comment]
@yurivict Thank you for reporting this issue. What version of FreeBSD are you using, and what are you seeing—errors, etc.—as a result of the exceeded limit?

**Comment by kevinushey** [comment]
We use [CMake](https://cmake.org/) for building the C++ sources, and [ant](https://ant.apache.org/) for compiling the Java sources. Presumedly one of those two compilers is opening a large number of files. It would help if you could tell us _what_ files are being opened.

You might also want to ensure the RStudio components are built in serial rather than parallel if you aren't already.

**Comment by yurivict** [comment]
The problem exists on FreeBSD 11 and 12.

Error log: http://beefy10.nyi.freebsd.org/data/112i386-default/493733/logs/RStudio-1.2.679_8.log

The failure occurs when the system-level open file limit is exceeded, when the next file is opened. Increasing the limit solves the problem, but it shound't open so many files in the first place.

There isn't an easy way to determine the set of files.

**Comment by yurivict** [comment]
In order to reproduce, on linux you can set the limit on opened files, ex. 1000 should suffice for any build, and observe the failure. I suspect you build with a jacked up limit, like 10k.

**Comment by yurivict** [comment]
It keeps failing, see http://beefy4.nyi.freebsd.org/data/120i386-quarterly/500840/logs/errors/RStudio-1.2.1335.log
The package is not available now on FreeBSD for this reason.

**Comment by gtritchie** [comment]
One thing for us to look at is the number of images we have in our ClientBundles. https://groups.google.com/forum/?fromgroups=#!topic/google-web-toolkit/EGiOO0g85rI

No idea if this is the cause. We haven't intentionally done anything to increase our open-files during build time. If you are building master, we have updated to the latest Gwt, perhaps that has brought in some unfortunate side effects.

**Comment by yurivict** [comment]
Yes, I use the github master. This problem existed for a long while, but recently it got worse.

**Comment by yurivict** [comment]
ping!
